lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -1422,7 +1422,7 @@ check_terminator({End, {EndLine, EndColumn, _}}, [{Start, {StartLine, _, _}, _}
           "\n\n    HINT: the \"~ts\" on line ~B is missing terminator \"~ts\"",
           [Start, StartLine, ExpectedEnd]
         ),
-      {error, {EndLine, EndColumn, {unexpected_token_or_reserved(End), Suffix}, [atom_to_list(End)]}}
+      {error, {S, E, {unexpected_token_or_reserved(End), Suffix}, [atom_to_list(End)]}}
```

Comment:
Unfortunately this will likely push complexity to other parts of the stack, because the tuples are a new return type that do not match anywhere else in the code base. Unfortunately, I will have to ask you to do another large-ish change. You can see that in elixir.erl, we convert Line and Column into a keyword list:

```
    {error, {Line, Column, {ErrorPrefix, ErrorSuffix}, Token}, _Rest, _Warnings, _SoFar} ->
      Location = [{line, Line}, {column, Column}],
      {error, {Location, {to_binary(ErrorPrefix), to_binary(ErrorSuffix)}, to_binary(Token)}};
    {error, {Line, Column, Error, Token}, _Rest, _Warnings, _SoFar} ->
      Location = [{line, Line}, {column, Column}],
      {error, {Location, to_binary(Error), to_binary(Token)}}
```

My suggestion is to change elixir_tokenizer.erl to return `{error, {Location, Error | {Prefix, Suffix}, Token}, ...}`. In other words, instead of returning Line and Column separately, we already build the location! This way, when you build the mismatched delimiter error, you can add line, column, end_line, and end_column. WDYT? Also please make sure that it is always `line` and then `column`, as you will need to match on them in `elixir_interpolation.erl`. You can have an Erlang macro called `?LOC(Line, Column)` to help you build it. :)

Reply:
Thanks for the guidance! I'll take a better look tonight

But I'm not sure I got it 100% from the `error, {Location, Error | {Prefix, Suffix}, Token}, ...}` example, the idea is for us to return two locations as in `{StartLine, StartCol}, {EndLine, EndCol}` and then update the code you showed to extract line and column, and also add end_line and end_col if existing to the keyword list? 

Since we're gonna only have an end position on the mismatched delimiter cases we would return the end location as nil in other cases, I assume. Is that right?

Reply:
The idea is for Location to be a keyword list, so it doesn't matter how many entries it has. This way, we no longer need to convert it to a keyword list in elixir.erl.